libcaf_core/src/telemetry/metric_registry.cpp

```diff
@@ -35,6 +36,21 @@ metric_registry::~metric_registry() {
   // nop
 }
 
+void metric_registry::merge(metric_registry& other) {
+  if (this == &other)
+    return;
+  std::unique_lock<std::mutex> guard1{families_mx_};
+  std::unique_lock<std::mutex> guard2{other.families_mx_};
```

Comment:
Shouldn't this be `auto lock = std::scoped_lock{families_mx_, other.families_mx_};`?

Reply:
No, we always use this notation for scope guards.

Reply:
More in the sense of you need to lock both at the same time using either `std::lock(...)` with `std::{defer,adopt}_lock` or use the C++17 replacement for `std::unique_lock`, which is `std::scoped_lock`.

Reply:
Switching to the "C++17 style" lock guards certainly is a good idea. I just wouldn't like them as a random one off in the code base via this PR.

Reply:
Wit C++11 you were still able to do this:

```suggestion
  std::unique_lock<std::mutex> guard1{families_mx_, std::defer_lock};
  std::unique_lock<std::mutex> guard2{other.families_mx_, std::defer_lock};
  std::lock(guard1, guard2);
```

Reply:
Are you suggesting this for the deadlock avoidance by `std::lock`? This is the only piece of code that touches both registries at once and there aren't any other algorithms or use cases that operate on two metric registries at once (this just isn't a thing), so I don't see a need for try-and-back-off or other avoidance strategies that `std::lock` may apply.

Switching to the new `scoped_lock` for cleaner code is a good idea, but rather as a refactoring step on the entire code base.